src/Core/Autoload/LegacyClassLoader.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Autoload;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Exception\DirectoryNotFoundException;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+final class LegacyClassLoader
+{
+    public const CORE_SUFFIX = 'Core';
+
+    private $coreDirectories = ['classes', 'controllers'];
+    private $overrideDirectories = ['override/Classes', 'override/Controllers'];
+    private $filesystem;
+
+    /**
+     * @var string
+     */
+    private $rootDirectory;
+
+    /**
+     * @var string
+     */
+    private $cacheDirectory;
+
+    public function __construct(string $directory, string $cacheDirectory)
+    {
+        if (!str_ends_with($directory, DIRECTORY_SEPARATOR)) {
```

Comment:
str_ends_with is available from php8. If I'm not mistaken Prestashop 8.1 should also be compatible with 7.4  and maybe less?

Reply:
up to php 7.2 yes ;) . But there is the polyfill given by Symfony that provides the function compatibility for us.